quantum/rgblight.c

```diff
@@ -118,6 +119,35 @@ void setrgb(uint8_t r, uint8_t g, uint8_t b, LED_TYPE *led1) {
   (*led1).b = b;
 }
 
+void rgblight_check_config(void) {
+  /* Add some out of bound checks for RGB light config */
+
+  if (rgblight_config.mode < RGBLIGHT_MODE_STATIC_LIGHT) {
+    rgblight_config.mode = RGBLIGHT_MODE_STATIC_LIGHT;
+  }
+  else if (rgblight_config.mode > RGBLIGHT_MODES) {
+    rgblight_config.mode = RGBLIGHT_MODES;
+  }
+
+  if (rgblight_config.hue < 0) {
+    rgblight_config.hue = 0;
+  } else if (rgblight_config.hue > 360) {
```

Comment:
I'd prefer some #defines to avoid some of these magic numbers. 

```
#define RGBLIGHT_HUE_MIN 0
#define RGBLIGHT_HUE_MAX 360
#define RGBLIGHT_SAT_MIN 0
#define RGBLIGHT_SAT_MAX 255
```

Reply:
I don't think your comment disagrees with my statement. 

My main argument is that rather than using the number 360, explicitly #define it as the max.  Same goes for 255. Why do you not think we should #define these? I think it's a lot more readable. And what limiting are you referring to in your comment? I don't think making a #define called MAX for 360 limits it anymore than the fact that 360 is the max.

Reply:
@jackhumbert or @skullydazed, do you have any input on this?